lib/libc/riscv/SYS.h

```diff
@@ -42,6 +42,24 @@
 	li	t0, SYS_ ## name;				\
 	ecall
 
+#ifndef __CHERI_PURE_CAPABILITY__
+#define _GET_FNPTR(outreg, function)	la outreg, _C_LABEL(function)
+#define _CALL_FNPTR(fnptr)	jalr cra, fnptr
+#define _TAILCALL_FNPTR(fnptr)	jr fnptr
+#define _CALL_TMPREG	t1
+#else
+#define _GET_FNPTR(outreg, function)	clgc outreg, _C_LABEL(function)
+#define _CALL_FNPTR(fnptr)	cjalr cra, fnptr
+#define _TAILCALL_FNPTR(fnptr)	cjr fnptr
+#define _CALL_TMPREG	ct1
+#endif
+#define ASM_TAILCALL(function)					\
+	_GET_FNPTR(_CALL_TMPREG, function);			\
+	_TAILCALL_FNPTR(_CALL_TMPREG)
+#define ASM_CALL(function)					\
```

Comment:
ASM_CALL doesn't seem to be used?

Note that you could permit 't1' to be specified as a parameter and use `c ## reg` in the purecap version of the macro to convert it to a cap register.

Reply:
Yes that sounds good.
I added the CALL version since I think I'll need it in RTLD.